File: server/Models/GreenhouseFloor.cs
namespace Server.Models;

/// <summary>
/// One level of a <see cref="Greenhouse"/>'s layout — a greenhouse can have several, each with its
/// own set of <see cref="GreenhouseSection"/>s. Most greenhouses have exactly one; multiple floors
/// exist for the (rarer) multi-level structure.
/// </summary>
public class GreenhouseFloor
{
    public int Id { get; set; }
    public int GreenhouseId { get; set; }
    public string Name { get; set; } = string.Empty;

    /// <summary>Display order in the floor list/switcher — floors are added at the end by default.</summary>
    public int SortOrder { get; set; }
}

File: web/src/types/greenhouse-floor.ts
/** One level of a greenhouse's layout — most greenhouses have exactly one; a multi-level
 * structure has several, each with its own set of sections. */
export type GreenhouseFloor = {
  id: number;
  greenhouseId: number;
  name: string;
  /** Display order in the floor switcher. Assigned by the server; not client-editable. */
  sortOrder: number;
};

export type GreenhouseFloorInput = Omit<GreenhouseFloor, 'id' | 'sortOrder'>;
